### ISSUE TITLE
`colorProvider: null` does not conform to specification

### ISSUE THREAD

**Issue by kdheepak** [autozimu/LanguageClient-neovim#940]
- Did you upgrade to latest plugin version?

Yes.

- Did you upgrade to/compile latest binary? Run shell command
  `bin/languageclient --version` to get its version number.

Yes.

- (Neovim users only) Did you check output of `:checkhealth LanguageClient`?

```
health#LanguageClient#check
========================================================================
  - OK: binary found: /Users/USER/.local/share/nvim/plugged/LanguageClient-neovim/bin/languageclient
  - OK: languageclient 0.1.155 1ffab7ef19e2acae6f32c6f3a27581429d28e47b
  - OK: Floating window is supported and will be used for hover
```
                                                                        
- Did you check [troubleshooting]?

Yes. I get the same error with a `min-vimrc.vim`. I have shared the configuration below.

[troubleshooting]: https://github.com/autozimu/LanguageClient-neovim/blob/next/INSTALL.md#6-troubleshooting

## Describe the bug

Using Julia and neovim, with the following configuration:

The `LanguageClient-neovim` plugin sends the following message to the Julia LanguageServer:

```
14:06:19 INFO writer-Some("julia") src/rpcclient.rs:215 => Some("julia") {"jsonrpc":"2.0","method":"initialize","params":{"capabilities":{"textDocument":{"colorProvider":null,"completion":{"completionItem":{"snippetSupport":false}},"declaration":{"linkSupport":true},"definition":{"linkSupport":true},"implementation":{"linkSupport":true},"publishDiagnostics":{"relatedInformation":true},"signatureHelp":{"signatureInformation":{"parameterInformation":{"labelOffsetSupport":true}}},"typeDefinition":{"linkSupport":true}},"workspace":{"applyEdit":true,"didChangeWatchedFiles":{"dynamicRegistration":true}}},"processId":77091,"rootPath":"/Users/USER/GitRepos/Test","rootUri":"file:///Users/USER/GitRepos/Test","trace":"off"},"id":1}
```

`colorProvider` is set to `null` in this message. [According to the specification](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/), `colorProvider` can either be not defined or a boolean value.

This causes the following error: 

```
┌ Warning: no Manifest.toml file found, static paths used
└ @ Revise ~/.julia/packages/Revise/S7mrl/src/Revise.jl:1190
[ Info: Started symbol server
[ Info: Tried to load bac558e1-5e72-5ebc-8fee-abe8a469f55d but failed to find it in the manifest.
[ Info: Tried to load 37e2e46d-f89d-539d-b4ee-838fcccc9c8e but failed to find it in the manifest.
[ Info: Tried to load 2f01184e-e22b-5df5-ae63-d93ebab69eaf but failed to find it in the manifest.
[ Info: Tried to load 8f399da3-3557-5675-b5ff-fb832c97cbdb but failed to find it in the manifest.
[ Info: Tried to load b8f27783-ece8-5eb3-8dc8-9495eed66fee but failed to find it in the manifest.
[ Info: Tried to load fdba3010-5040-5b88-9595-932c9decdf73 but failed to find it in the manifest.
[ Info: store set in 0.014803887 seconds
[ Info: Dict{String,Any}("workspace" => Dict{String,Any}("didChangeWatchedFiles" => Dict{String,Any}("dynamicRegistration" => true),"applyEdit" => true),"textDocument" => Dict{String,Any}("definition" => Dict{String,Any}("linkSupport" => true),"declaration" => Dict{String,Any}("linkSupport" => true),"publishDiagnostics" => Dict{String,Any}("relatedInformation" => true),"completion" => Dict{String,Any}("completionItem" => Dict{String,Any}("snippetSupport" => false)),"implementation" => Dict{String,Any}("linkSupport" => true),"signatureHelp" => Dict{String,Any}("signatureInformation" => Dict{String,Any}("parameterInformation" => Dict{String,Any}("labelOffsetSupport" => true))),"typeDefinition" => Dict{String,Any}("linkSupport" => true),"colorProvider" => nothing))
ERROR: MethodError: Cannot `convert` an object of type Nothing to an object of type Bool
Closest candidates are:
  convert(::Type{T}, !Matched::T) where T<:Number at number.jl:6
  convert(::Type{T}, !Matched::Number) where T<:Number at number.jl:7
  convert(::Type{T}, !Matched::Ptr) where T<:Integer at pointer.jl:23
  ...
Stacktrace:
 [1] convert(::Type{Union{Missing, Bool}}, ::Nothing) at ./missing.jl:69
 [2] LanguageServer.Capabilities(::Nothing) at /Users/USER/.julia/dev/LanguageServer/src/protocol/initialize.jl:4
 [3] LanguageServer.TextDocumentClientCapabilities(::Dict{String,Any}) at /Users/USER/.julia/dev/LanguageServer/src/protocol/protocol.jl:37
 [4] LanguageServer.ClientCapabilities(::Dict{String,Any}) at /Users/USER/.julia/dev/LanguageServer/src/protocol/protocol.jl:37
 [5] LanguageServer.InitializeParams(::Dict{String,Any}) at /Users/USER/.julia/dev/LanguageServer/src/protocol/initialize.jl:183
 [6] parse_params(::Type{Val{:initialize}}, ::Dict{String,Any}) at /Users/USER/.julia/dev/LanguageServer/src/requests/init.jl:73
 [7] parse(::Type{LanguageServer.JSONRPC.Request}, ::Dict{String,Any}) at /Users/USER/.julia/dev/LanguageServer/src/jsonrpc.jl:46
 [8] run(::LanguageServerInstance) at /Users/USER/.julia/dev/LanguageServer/src/languageserverinstance.jl:101
 [9] top-level scope at none:1
/Users/USER/.julia/dev/LanguageServer/contrib/languageserver.sh: line 34: kill: (77142) - No such process
```

[LanguageServer.jl](https://github.com/julia-vscode/LanguageServer.jl) conforms to the specification strictly. See discussion  here for more information: https://github.com/julia-vscode/LanguageServer.jl/pull/454

## Environment
- neovim/vim version (`nvim --version` or `vim --version`):

```
$ nvim --version
NVIM v0.5.0-279-ge922576bd
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: /Applications/Xcode-10.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/Users/travis/build/neovim/bot-ci/build/neovim/build/config -I/Users/travis/build/neovim/bot-ci/build/neovim/src -I/Users/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/usr/local/opt/gettext/include -I/usr/include -I/Users/travis/build/neovim/bot-ci/build/neovim/build/src/nvim/auto -I/Users/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@Traviss-Mac-6.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/share/nvim"

Run :checkhealth for more info
```

- This plugin version (`git rev-parse --short HEAD`):

```
$ cd ~/.local/share/nvim/plugged/LanguageClient-neovim
$ git rev-parse --short HEAD
0394fba
```

- This plugin's binary version (`bin/languageclient --version`):

```
$ ~/.local/share/nvim/plugged/LanguageClient-neovim/bin/languageclient --version
languageclient 0.1.155 1ffab7ef19e2acae6f32c6f3a27581429d28e47b
```

- Minimal vimrc content (A minimal vimrc is the smallest vimrc that could
  reproduce the issue. Refer to an example [here][min-vimrc.vim]):

```
call plug#begin('~/.local/share/nvim/plugged')

Plug 'JuliaEditorSupport/julia-vim'
Plug 'autozimu/LanguageClient-neovim', {
    \ 'branch': 'next',
    \ 'do': 'bash install.sh',
    \ }

call plug#end()

" Always draw sign column. Prevent buffer moving when adding/deleting sign.
set signcolumn=yes

let g:LanguageClient_serverCommands = {
    \ 'julia': ['~/Applications/Julia-1.3.app/Contents/Resources/julia/bin/julia', '--startup-file=no', '--history-file=no', '--project', '-e', '
    \       using LanguageServer;
    \       using Pkg;
    \       env_path = dirname(Pkg.Types.Context().env.project_file);
    \       debug = false;
    \       server = LanguageServer.LanguageServerInstance(stdin, stdout, debug, env_path, "");
    \       server.runlinter = true;
    \       run(server);
    \   '],
    \ }

let g:LanguageClient_loggingLevel = 'INFO'
let g:LanguageClient_virtualTextPrefix = ''
let g:LanguageClient_loggingFile =  expand('~/.local/share/nvim/LanguageClient.log')
let g:LanguageClient_serverStderr = expand('~/.local/share/nvim/LanguageServer.log')
```

- Language server link and version:

https://github.com/julia-vscode/LanguageServer.jl/tree/b9ba36fc96801671752ba15242a3d4fc2e8d25be

[min-vimrc.vim]: https://github.com/autozimu/LanguageClient-neovim/blob/next/min-vimrc.vim

## To Reproduce
1. Store settings above in min-vimrc.vim
1. Start vim: `nvim -u min-vimrc.vim test.jl`
1. See error in `~/.local/share/nvim/LanguageServer.log`

## Current behavior

The current behaviour sends `colorProvider: null` as part of the JSON, and that does not match the spec.

## Expected behavior

Send `colorProvider: false` instead, or don't send a key value pair for the key `colorProvider` if the value is `null`.

**Comment by ExpandingMan** [comment]
Just wanted to bump this in case anyone is watching.  This issue is crashing our Julia language server as well.

**Comment by martskins** [comment]
This seems to be an issue with a missing attribute in the crate `lsp-types`. Just opened https://github.com/gluon-lang/lsp-types/pull/160 to fix that, it should be fixed once that gets merged and we upgrade the dependency.

**Comment by martskins** [comment]
@ExpandingMan Opened #1041, if/once that gets merged you should be able to verify that this is working correctly in the `dev` branch.

**Comment by ExpandingMan** [comment]
Nice, thanks.  There seem to be a multitude of other issues (I think on both server and client side), but I'll definitely try it out and see what happens when this gets merged.

**Comment by martskins** [comment]
Make sure to open issues for those, hopefully someone will be able to provide an answer for the client side ones at least. :pray:

**Comment by martskins** [comment]
Closing this, as it is fixed by #1041.
